File: tools/tal_support/as274_fc/C/t5.c
#include <stdio.h>
#include "as274c.h"
#include "misc.h"

int main()
{
    double *d, *rbar, *rinv, rho=.6, *covmat, var, *sterr;
    int pos=0, row, col, error;

    d = dvector(0, 6);
    rbar = dvector(0, 20);
    rinv = dvector(0, 14);
    covmat = dvector(0, 20);
    sterr = dvector(0, 5);

    for (row=0; row<6; row++) {
        d[row] = 1;
        for (col=row+1; col<7; col++)
          rbar[pos++] = col == row+1 ? rho : 0;
    }

    var = 1;
    error = cov(7, 21, d, rbar, 6, rinv, &var, covmat, 21, sterr);
    if (error) printf("cov() returned %d\n", error);
    printf("covmat is:\n");
    pr_utdm_v(covmat, 6, 13, 7);
    return 0;
}

File: tools/tal_support/as274_fc/F/t5.f

      program test5
c
c     Test of subroutine COV.
c
      double precision d(7), rbar(21), rinv(15), one, rho
      double precision zero, covmat(21), var,
     +      sterr(6)
      integer pos, row, col, ier
c
      data one/1.0/, rho/0.6/, zero/0.0/
c
      pos = 1
      do 20 row = 1, 6
	d(row) = one
	do 10 col = row+1, 7
	  if (col .eq. row+1) then
	    rbar(pos) = rho
	  else
	    rbar(pos) = zero
	  end if
	  pos = pos + 1
   10   continue
   20 continue
      var = one
      call cov(7, 21, d, rbar, 6, rinv, var, covmat, 21, sterr, ier)
      write(*, 900) covmat
  900 format(1x, 6f12.6/ 13x, 5f12.6/ 25x, 4f12.6/ 37x, 3f12.6/
     +      49x, 2f12.6/ 61x, f12.6)
      end
